File: CareSync.BusinessAPI/Entities/Doctor.cs
// using CareSync.BusinessAPI.Enums;

namespace CareSync.BusinessAPI.Entities;

public class Doctor : BaseEntity
{
    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Phone { get; set; } = string.Empty;

    public string Gender { get; set; } = string.Empty;

    public string Specialization { get; set; } = string.Empty;

    public string Qualification { get; set; } = string.Empty;

    public int Experience { get; set; }

    public string RoomNumber { get; set; } = string.Empty;

    public bool IsAvailable { get; set; } = true;

    public Guid DepartmentId { get; set; }

    public Department? Department { get; set; }
}

File: frontend/src/types/Doctor.ts
export interface Doctor {
  id: string;

  firstName: string;

  lastName: string;

  email: string;

  phone: string;

  gender: string;

  specialization: string;

  qualification: string;

  experience: number;

  roomNumber: string;

  isAvailable: boolean;

  departmentId: string;
}

export interface DoctorApiResponse {
  success: boolean;

  message: string;

  data: Doctor[];
}

export interface DoctorResponse {
  success: boolean;

  message: string;

  data: Doctor;
}

export interface CreateDoctorRequest {
  firstName: string;

  lastName: string;

  email: string;

  phone: string;

  gender: string;

  specialization: string;

  qualification: string;

  experience: number;

  roomNumber: string;

  isAvailable: boolean;

  departmentId: string;
}

export interface UpdateDoctorRequest {
  firstName: string;

  lastName: string;

  email: string;

  phone: string;

  gender: string;

  specialization: string;

  qualification: string;

  experience: number;

  roomNumber: string;

  isAvailable: boolean;

  departmentId: string;
}